av/container/input.pyx

```diff
@@ -65,7 +70,18 @@ cdef class InputContainer(Container):
 
         self.streams = StreamContainer()
         for i in range(self.ptr.nb_streams):
-            self.streams.add_stream(wrap_stream(self, self.ptr.streams[i]))
+            stream = self.ptr.streams[i]
+            codec = lib.avcodec_find_decoder(stream.codecpar.codec_id)
+            if codec:
+                # allocate and initialise decoder
+                codec_context = lib.avcodec_alloc_context3(codec)
+                err_check(lib.avcodec_parameters_to_context(codec_context, stream.codecpar))
```

Comment:
`avcodec_parameters_to_context` does not set the `framerate` field of `AVCodecContext`: https://www.ffmpeg.org/doxygen/trunk/codec__par_8c_source.html#l00182
As `Stream.rate`/`Stream.framerate` relies on getattr pass through to its codec_context's rate/framerate, this means `Stream.rate`/`Stream.framerate` are no longer valid for input video streams.
Two ways we can remedy this:
1) We can set the codec_context's framerate here and continue to rely on this passthrough feature.
2) We can add `rate`/`framerate` properties directly to `VideoStream`.
Note that `AudioStream` does not have this issue as the `sample_rate` field gets copied properly.

Reply:
I'm not even sure we should be doing anything:

https://www.ffmpeg.org/doxygen/trunk/structAVCodecContext.html#a4d08b297e97eefd66c714df4fff493c8

`decoding: For codecs that store a framerate value in the compressed bitstream, the decoder may export it here. { 0, 1} when unknown.`

Reply:
I'm pretty sure it's not always set. It was causing a failure in our test suite that I remedied using the 2nd option above.
However, what you said actually implies we should rely on the first option above (passthrough), because the decoder may export it when decoding the bitstream.